parser/ndt.go

```diff
@@ -149,22 +151,24 @@ func (n *NDTParser) ParseAndInsert(taskInfo map[string]bigquery.Value, testName
 		return nil
 	}
 
-	taskFileName := taskInfo["filename"].(string)
-
 	if info.Time != n.timestamp {
-		// All files are processed ASAP.  However, if there is ONLY
-		// a data file, or ONLY a meta file, we have to process the
-		// test files anyway.
-		// TODO(dev) Handle case where we don't get a meta file on the last
-		// test in a task.
-		n.handleAnomolies(taskFileName)
+		// Handle previous test group before processing new group.
+		n.processGroup()
 
+		n.taskFileName = taskInfo["filename"].(string)
 		n.timestamp = info.Time
-		n.s2c = nil
-		n.c2s = nil
-		n.metaFile = nil
+	} else {
+		// Within a group of tests, we expect consistent taskInfo.
+		if n.taskFileName != taskInfo["filename"].(string) {
+			metrics.TestCount.WithLabelValues(
+				n.TableName(), "any", "inconsistent taskFileName").Inc()
+		}
 	}
 
+	// Because of port number, the c2s, s2c, and meta files may come in
+	// any order.  We defer processing until Flush or new test group.
+	// TODO - should we just ignore non-gzipped test files?  Or do some archives
+	// have unzipped files?
```

Comment:
It looks like gzip compression capabilities did not arrive until after MLab started ( https://github.com/ndt-project/ndt/blob/master/src/logging.c#L45 ). So I can confidently say that ignoring all non-zipped results is the wrong thing.  Ignoring unzipped files if the zipped ones are present is definitely the right thing.